src/components/Form/Form.tsx

```diff
@@ -3,47 +3,89 @@ import {
 	StyledDiv,
 	StyledFieldset,
 	StyledInput,
+	StyledLabel,
 	StyledLegend,
 } from "./Form.Styled.ts";
 import { useState } from "react";
 import NavLinkButton from "../Button/NavLinkButton.tsx";
+import InputError from "./InputError.tsx";
+import { useNavigate } from "react-router-dom";
 
 function LoginForm() {
 	const [password, setPassword] = useState("");
-	const [name, setName] = useState("");
+	const [email, setEmail] = useState("");
+	const [error, setError] = useState<{ email?: string; password?: string }>({});
+	const navigate = useNavigate();
+
+	const validateEmail = (email: string) => {
+		const validEmail = new RegExp(
+			"^[a-zA-Z0-9._:$!%-]+@[a-zA-Z0-9.-]+.[a-zA-Z]$"
+		);
+		return validEmail.test(email);
+	};
+
+	const validatePassword = (password: string) => {
+		return password.length >= 6;
+	};
+
+	const handleSubmit = (e: React.FormEvent) => {
+		console.log(e.target);
+		e.preventDefault();
+		const newError: { email?: string; password?: string } = {};
+
+		if (!validateEmail(email)) {
+			newError.email = "Email format is wrong";
```

Comment:
będzie prościej: setError({ ...error, email: "Email format is wrong" }); i od razu pod tym `return`  który nam zakonczy funkcje.

Dla passwordu analogicznie, tylko zamiast email, to password wiadomo.
Wtedy object.keys będzie zbędny.